ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -3195,17 +3196,24 @@ public void alterPartitions(String dbName, String tableName,
   public List<org.apache.hadoop.hive.metastore.api.Partition> getPartitionsByNames(String dbName, String tableName,
       List<String> partitionNames) throws HiveException {
     try {
-      return getMSC().getPartitionsByNames(dbName, tableName, partitionNames);
+      GetPartitionsByNamesRequest req = new GetPartitionsByNamesRequest();
+      req.setDb_name(dbName);
+      req.setTbl_name(tableName);
+      req.setNames(partitionNames);
+      return getPartitionsByNames(req, null);
     } catch (Exception e) {
       LOG.error("Failed getPartitionsByNames", e);
       throw new HiveException(e);
     }
   }
 
-    public List<org.apache.hadoop.hive.metastore.api.Partition> getPartitionsByNames(GetPartitionsByNamesRequest req)
+    public List<org.apache.hadoop.hive.metastore.api.Partition> getPartitionsByNames(GetPartitionsByNamesRequest req,
+      Table table)
         throws HiveException {
     try {
-      Table table = getTable(req.getDb_name(), req.getTbl_name());
+      if( table == null ) {
+        table = getTable(req.getDb_name(), req.getTbl_name());
```

Comment:
This can cause a round trip to HMS to get the table too for none cached client. I think you can add the logic here: https://github.com/apache/hive/blob/master/standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClientWithLocalCache.java#L460

Reply:
Table Id is not cached right now in the local cache, so it would be the same.

Reply:
As if you do not use cached client, you need not table id, it will be different for the client direct use HMS apis (no need to call getTable).

Reply:
TableId is still required for the remote cache to decide whether it can serve the data from cache or has to refresh it from HMS.

Reply:
Ok, removed the extra getTable call now.